mydumper.c

```diff
@@ -1066,10 +1068,21 @@ MYSQL *create_main_connection()
 
 	// We should not set a DB in TiDB until after a tidb_snapshot is set.
 	// This preserves the use-case of restoring a dropped database
-	if ((detected_server != SERVER_TYPE_TIDB) && db) {
-		g_message("Selecting database: %s", db);
-		mysql_select_db(conn, db);
-	}
+    if ((detected_server != SERVER_TYPE_TIDB) && db) {
+        g_message("Selecting database: %s", db);
+        mysql_select_db(conn, db);
+    }
+
+    if (tidb_force_priority) {
+        if (detected_server == SERVER_TYPE_TIDB) {
+            char *query = g_strdup_printf("SET SESSION tidb_force_priority = \"%s\"", tidb_force_priority);
```

Comment:
```suggestion
            char *query = g_strdup_printf("SET SESSION tidb_force_priority = '%s'", tidb_force_priority);
```

*Always* use single quotes for strings to avoid dealing with ANSI_QUOTES.